dpnp/linalg/dpnp_utils_linalg.py

```diff
@@ -27,17 +27,60 @@
 # *****************************************************************************
 
 
+import dpctl
 import dpctl.tensor._tensor_impl as ti
+import dpctl.utils as du
+from numpy import prod
 
 import dpnp
 import dpnp.backend.extensions.lapack._lapack_impl as li
 
-__all__ = ["dpnp_eigh"]
+__all__ = ["dpnp_eigh", "dpnp_solve"]
 
 _jobz = {"N": 0, "V": 1}
 _upper_lower = {"U": 0, "L": 1}
 
 
+def _common_type(*arrays):
+    """
+    _common_type(*arrays)
+
+    Common type for linalg
+
+    This function determines the common data type for linalg operations.
+    It's designed to be similar in logic to `numpy.linalg.linalg._commonType`.
+
+    Key differences from `numpy.common_type`:
+    - It accepts ``bool_`` arrays.
+    - The default floating-point data type is determined by the capabilities of the device,
+      as indicated by `dpnp.default_float_type()`.
+
+    Args:
+        *arrays (dpnp.ndarray): Input arrays.
+
+    Returns:
+        dtype_common (dtype): The common data type for linalg operations.
+
+        This returned value is applicable both as the precision to be used
+        in linalg calls and as the dtype of (possibly complex) output(s).
+
+    """
+
+    dtypes = [arr.dtype for arr in arrays]
+
+    default = dpnp.default_float_type().name
```

Comment:
It seems `dpnp.default_float_type()` returns the default float type on `gpu`. We need to path the input array device to it to get the correct float type when the input arrays arrays are on `cpu`.